### PR TITLE
[CICD-8022] Create job for first time auto-bootstrap

### PR DESCRIPTION
**Pull request by jnathan-ias**
Create a jenkins job `.xml` definition and load it during the autoconfig process during `init-hook` phase

Summary
---------
Define a jenkins job in `.xml` to be uploaded and added to the jenkins controller in the `init-hook` phase during first boot. 
Job should allow input from the administrator to pass their github username as `JENKINS_USER` and github personal access token as `JENKINS_PASSWORD`. Job should be added to the jenkins controller and built with parameters. 

AC
--------
- jenkins first boot job is created via `.xml` definition 
- job is added to the jenkins controller
- job is added after credentials are configured
- job should provide build parameters so administrators can supply `JENKINS_USER` and `JENKINS_PASSWORD` values

See Also
-------
- [Jira]

[Jira]:https://jira.integralads.com/browse/CICD-8022

### CODE REVIEW

File: scripts/init.groovy.d/__bootstrap_jenkins_autoconfig.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        <hudson.model.StringParameterDefinition>
+          <name>JENKINS_USER</name>
+          <description>Github username of admin</description>
+          <defaultValue></defaultValue>
+          <trim>false</trim>
+        </hudson.model.StringParameterDefinition>
+        <hudson.model.PasswordParameterDefinition>
+          <name>JENKINS_PASSWORD</name>
+          <description>Administrators github Personal Access Token with scopes (user:email, repo, read:org)</description>
+          <defaultValue>{AQAAABAAAABAIhhGLd2ww/D30Fh5V0Tb02PKJiOvpaEnU0Cu3wzxo7Z0JzxLoN8uoRMcW12k16ppPP5KVh86hAJkmv/DkLVHX6fhSkHQMWKMZcYpNYtcx+8=}</defaultValue>
```
Comment (eshackelford-ias) [inline scripts/init.groovy.d/__bootstrap_jenkins_autoconfig.xml:21]: Is it necessary to have this `defaultValue` here? Unsure how sensitive this is, but just noting that anyone with access to the Jenkins controller or script console could decrypt this. I think that's just our team though, so this probably isn't a blocker.

File: scripts/init.groovy.d/aab_apply_autoconfig_job.groovy [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    println 'Skipping Jenkins auto-configuration'
+    return
+}
+
+/**
+  * From this point onward Jenkins should configure itself.
+  */
+
+import jenkins.model.Jenkins
+import groovy.xml.*
```
Comment (eshackelford-ias) [inline scripts/init.groovy.d/aab_apply_autoconfig_job.groovy]: Suggestion (nit):
* Import necessary XML classes individually, only those that are required.


For example:

```
import groovy.xml.XMLParser
import groovy.xml.XMLSlurper  // if necessary — looks like we're only parsing here though
```

This feedback is non-blocking.

File: scripts/init.groovy.d/__bootstrap_jenkins_autoconfig.xml (hunk 1 above)
Comment (eshackelford-ias) [inline scripts/init.groovy.d/__bootstrap_jenkins_autoconfig.xml:21]: Is it necessary to have this `defaultValue` here? Unsure how sensitive this is, but just noting that anyone with access to the Jenkins controller or script console could decrypt this. I think that's just our team though, so this probably isn't a blocker.

**Review by eshackelford-ias — APPROVED** [review]
LGTM, left 2 comments but non-blocking

File: scripts/init.groovy.d/__bootstrap_jenkins_autoconfig.xml (hunk 1 above)
Comment (jnathan-ias) [reply scripts/init.groovy.d/__bootstrap_jenkins_autoconfig.xml:21]: The default value is just `CHANGEME` 
Might be better to not have it, might be best if I can make this a required parameter that needs to be filled if thats possible.

File: scripts/init.groovy.d/aab_apply_autoconfig_job.groovy (hunk 2 above)
Comment (jnathan-ias) [reply scripts/init.groovy.d/aab_apply_autoconfig_job.groovy]: Will implement 👍

File: scripts/init.groovy.d/__bootstrap_jenkins_autoconfig.xml (hunk 1 above)
Comment (eshackelford-ias) [reply scripts/init.groovy.d/__bootstrap_jenkins_autoconfig.xml:21]: We talked about this on Slack. It's good how it is!

**Review by eshackelford-ias — APPROVED** [review]
LGTM!

**Pull request merged**